src/query/geo.rs

```diff
@@ -162,38 +162,60 @@ impl Serialize for Type {
 /// Geo Distance query
 ///
 /// TODO: Specific full unit test for querying with a generated query from here
-#[derive(Debug, Default, Serialize)]
-pub struct GeoDistanceQuery {
-    field: String,
-    location: Location,
+#[derive(Debug, Serialize)]
+pub struct GeoDistanceQuery(FieldBasedQuery<Location, GeoDistanceQueryOuter>);
+
+#[derive(Debug, Default)]
+struct GeoDistanceQueryOuter {
     distance: Distance,
     distance_type: Option<DistanceType>,
     optimize_bbox: Option<OptimizeBbox>,
     coerce: Option<bool>,
     ignore_malformed: Option<bool>
 }
 
+impl MergeSerialize for GeoDistanceQueryOuter {
+    fn merge_serialize<S>(&self,
+                          serializer: &mut S) -> Result<(), S::Error>
+        where S: SerializeMap {
+
+        serializer.serialize_entry("distance", &self.distance)?;
+        if let Some(ref t) = self.distance_type {
+            serializer.serialize_entry("distance_type", t)?;
+        }
+        if let Some(ref b) = self.optimize_bbox {
+            serializer.serialize_entry("optimize_bbox", b)?;
+        }
```

Comment:
I've looked at the GeohashCellQueryOuter implementation that does not use this function. Maybe it also need some cosmetic changes.